awscli/customizations/configure/addmodel.py

```
@@ -0,0 +1,102 @@
+    # Get the mappings of endpoint prefixes to service names from the
+    # available service models.
+    prefixes_to_services = {}
+    for service_name in session.get_available_services():
+        service_model = session.get_service_model(service_name)
+        prefixes_to_services[service_model.endpoint_prefix] = service_name
+    return prefixes_to_services
+
+
+def get_model_location(session, service_definition, service_name=None):
+    """Gets the path of where a service-2.json file should go in ~/.aws/models
+
+    :type session: botocore.session.Session
+    :param session: A session object
+
+    :type service_definition: dict
+    :param service_definition: The json loaded service definition
+
+    :type service_name: str
+    :param service_name: The service name to use. If this not provided,
+        this will be determined from a combination of available services
+        and the service definition.
+
+    :returns: The path to where are model should be placed based on
+        the service defintion and the current services in botocore.
+    """
+    # Add the ServiceModel abstraction over the service json definition to
+    # make it easier to work with.
+    service_model = ServiceModel(service_definition)
+
+    # Determine the service_name if not provided
+    if service_name is None:
+        # Figure out names that do not match the endpoint prefi
+        name_mappings = _get_endpoint_prefix_to_name_mappings(session)
+        endpoint_prefix = service_model.endpoint_prefix
+        # Determine the service name
+        service_name = name_mappings.get(endpoint_prefix, endpoint_prefix)
+
+    api_version = service_model.api_version
+
+    # For the model location we only want the custom data path (~/.aws/models
+    # not the one set by AWS_DATA_PATH)
+    data_path = session.get_component('data_loader').CUSTOMER_DATA_PATH
+    return os.path.join(data_path, service_name, api_version, 'service-2.json')
```

Seems like we should try to pull the model version from the JSON file first and only fall back to a hardcoded default of 2 if the key does not exist.